tests/unit/package_managers/test_rpm.py

```diff
@@ -0,0 +1,279 @@
+from pathlib import Path
+from unittest import mock
+from urllib.parse import quote
+
+import pytest
+import yaml
+
+from cachi2.core.errors import PackageManagerError, PackageRejected
+from cachi2.core.models.sbom import Component
+from cachi2.core.package_managers.rpm import fetch_rpm_source, inject_files_post
+from cachi2.core.package_managers.rpm.main import (
+    DEFAULT_LOCKFILE_NAME,
+    DEFAULT_PACKAGE_DIR,
+    _createrepo,
+    _download,
+    _generate_repofiles,
+    _generate_repos,
+    _generate_sbom_components,
+    _resolve_rpm_project,
+    _verify_downloaded,
+)
+from cachi2.core.package_managers.rpm.redhat import RedhatRpmsLock
+from cachi2.core.rooted_path import RootedPath
+
+
+@pytest.fixture
+def rooted_tmp_path(tmp_path: Path) -> RootedPath:
+    return RootedPath(tmp_path)
+
+
+RPM_LOCK_FILE_DATA = """
+lockfileVersion: 1
+lockfileVendor: redhat
+arches:
+  - arch: x86_64
+    packages:
+      - url: https://example.com/x86_64/Packages/v/vim-enhanced-9.1.158-1.fc38.x86_64.rpm
+        checksum: sha256:21bb2a09852e75a693d277435c162e1a910835c53c3cee7636dd552d450ed0f1
+        size: 1976132
+        repoid: updates
+    source:
+      - url: https://example.com/source/tree/Packages/v/vim-9.1.158-1.fc38.src.rpm
+        checksum: sha256:94803b5e1ff601bf4009f223cb53037cdfa2fe559d90251bbe85a3a5bc6d2aab
+        size: 14735448
+        repoid: updates-source
+"""
+
+
+@mock.patch("cachi2.core.package_managers.rpm.main.RequestOutput.from_obj_list")
+@mock.patch("cachi2.core.package_managers.rpm.main._resolve_rpm_project")
+def test_fetch_rpm_source(
+    mock_resolve_rpm_project: mock.Mock,
+    mock_from_obj_list: mock.Mock,
+) -> None:
+    mock_component = mock.Mock()
+    mock_resolve_rpm_project.return_value = [mock_component]
+    mock_request = mock.Mock()
+    mock_request.rpm_packages = [mock.Mock()]
+    fetch_rpm_source(mock_request)
+    mock_resolve_rpm_project.assert_called_once()
+    mock_from_obj_list.assert_called_once_with(
+        components=[mock_component], environment_variables=[], project_files=[]
+    )
+
+
+def test_resolve_rpm_project_no_lockfile(rooted_tmp_path: RootedPath) -> None:
+    with pytest.raises(PackageRejected) as exc_info:
+        mock_source_dir = mock.Mock()
+        mock_source_dir.join_within_root.return_value.path.exists.return_value = False
+        _resolve_rpm_project(mock_source_dir, mock.Mock())
+    assert f"RPM lockfile '{DEFAULT_LOCKFILE_NAME}' missing, refusing to continue" in str(
+        exc_info.value
+    )
+
+
+@mock.patch(
+    "cachi2.core.package_managers.rpm.main.open",
```

Comment:
Can we pass an actual file in these tests and avoid mocking open and let yaml actually parse the file? These tests test the top-level function resolve so mocking everything isn't IMO acceptable.

Reply:
I removed mock open from tests that focused on parsing yaml/lockfile. 
Remained 2 other tests containing mock open. I hope it is not an issue.